qiskit_experiments/library/characterization/ramsey_xy.py

```diff
@@ -535,3 +544,280 @@ def _metadata(self) -> Dict[str, any]:
+                the rising and falling edges of the Stark tone.
+            stark_length (float): Time to accumulate Stark shifted phase in seconds.
+            min_stark_amp (float): Minimum Stark tone amplitude.
+            max_stark_amp (float): Maximum Stark tone amplitude.
+            num_stark_amps (int): Number of Stark tone amplitudes to scan.
+            stark_amps (list[float]): The list of amplitude that will be scanned in the experiment.
+                If not set, then ``num_stark_amps`` evenly spaced amplitudes
+                between ``min_stark_amp`` and ``max_stark_amp`` are used. If ``stark_amps``
+                is set, these parameters are ignored.
+        """
+        options = super()._default_experiment_options()
+        options.update_options(
+            stark_channel=None,
+            stark_freq_offset=80e6,
+            stark_sigma=15e-9,
+            stark_risefall=2,
+            stark_length=50e-9,
+            min_stark_amp=-1.0,
+            max_stark_amp=1.0,
+            num_stark_amps=101,
+            stark_amps=None,
+        )
+        options.set_validator("stark_freq_offset", (0, np.inf))
+        options.set_validator("stark_channel", pulse.channels.PulseChannel)
+        return options
+
+    def _set_backend(self, backend: Backend):
+        super()._set_backend(backend)
+        self._timing = BackendTiming(backend)
+
+    def parameters(self) -> np.ndarray:
+        """Stark tone amplitudes to use in circuits.
+
+        Returns:
+            The list of amplitudes to use for the different circuits based on the
+            experiment options.
+        """
+        opt = self.experiment_options  # alias
+
+        if opt.stark_amps is None:
+            params = np.linspace(opt.min_stark_amp, opt.max_stark_amp, opt.num_stark_amps)
+        else:
+            params = np.asarray(opt.stark_amps, dtype=float)
+
+        return np.round(params, decimals=7)
+
+    def parameterized_circuits(self) -> Tuple[QuantumCircuit, QuantumCircuit]:
+        """Create circuits with parameters for Ramsey XY experiment with Stark tone.
+
+        Returns:
+            Quantum template circuits for Ramsey X and Ramsey Y experiment.
+        """
+        opt = self.experiment_options  # alias
+        param = Parameter("stark_amp")
+        sym_param = param._symbol_expr
+
+        # Pulse gates
+        stark_v = Gate("StarkV", 1, [param])
+        stark_u = Gate("StarkU", 1, [param])
+
+        # Note that Stark tone yields negative (positive) frequency shift
+        # when the Stark tone frequency is higher (lower) than qubit f01 frequency.
+        # This choice gives positive frequency shift with positive Stark amplitude.
+        qubit_f01 = self._backend_data.drive_freqs[self.physical_qubits[0]]
+        neg_sign_of_amp = ParameterExpression(
+            symbol_map={param: sym_param},
+            expr=-sym.sign(sym_param),
+        )
+        abs_of_amp = ParameterExpression(
```

Comment:
This is supported very recently with https://github.com/Qiskit/qiskit-terra/pull/9309. I think this will be released in 0.25 but I don't want to block this PR for abs.

Reply:
Okay, I missed that it was so new. I will keep an eye on the sign issue you opened and we can try to remember to come back to class after it is addressed. We could fix this abs case before then but my main interest was in removing the accessing of the ParameterExpression's internal state and we need to address both the abs and the sign cases to accomplish that.